tests/unit-tests/integrations/maxmind-geolocation/class-wc-tests-maxmind-database.php

```diff
@@ -0,0 +1,135 @@
+<?php
+/**
+ * Class Functions.
+ *
+ * @package WooCommerce\Tests\Integrations
+ */
+
+/**
+ * Class WC_Tests_MaxMind_Database
+ */
+class WC_Tests_MaxMind_Database extends WC_Unit_Test_Case {
+
+	/**
+	 * Run setup code for unit tests.
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		// Callback used by WP_HTTP_TestCase to decide whether to perform HTTP requests or to provide a mocked response.
+		$this->http_responder = array( $this, 'mock_http_responses' );
+	}
+
+	/**
+	 * Tests that the database path filters work as intended.
+	 *
+	 * @expectedDeprecated woocommerce_geolocation_local_database_path
+	 */
+	public function test_database_path_filters() {
+		$path = WC_MaxMind_Geolocation_Database::get_database_path();
+		$this->assertEquals( WP_CONTENT_DIR . '/uploads/' . WC_MaxMind_Geolocation_Database::DATABASE . WC_MaxMind_Geolocation_Database::DATABASE_EXTENSION, $path );
+
+		add_filter( 'woocommerce_geolocation_local_database_path', array( $this, 'filter_database_path_deprecated' ), 1, 2 );
+		$path = WC_MaxMind_Geolocation_Database::get_database_path();
+		remove_filter( 'woocommerce_geolocation_local_database_path', array( $this, 'filter_database_path' ) );
+
+		$this->assertEquals( '/deprecated_filter', $path );
+
+		add_filter( 'woocommerce_geolocation_local_database_path', array( $this, 'filter_database_path' ) );
+		$path = WC_MaxMind_Geolocation_Database::get_database_path();
+		remove_filter( 'woocommerce_geolocation_local_database_path', array( $this, 'filter_database_path' ) );
+
+		$this->assertEquals( '/filter', $path );
+	}
+
+	/**
+	 * Tests that the database download works as expected.
+	 */
+	public function test_download_database_works() {
+		$expected_database = '/tmp/GeoLite2-Country_20200100/GeoLite2-Country.mmdb';
```

Comment:
This directory name may change with future releases of GeoLite and will make this test to fail.

Reply:
This test doesn't actually perform the download, but rather "downloads" a file from the `tests/data` directory. Since we can't distribute real MaxMind archives in the repository, this is a mock of the downloaded archive, with fixed file paths.

Reply:
Cool!